candle-examples/examples/falcon/main.rs

```diff
@@ -172,8 +172,10 @@ fn main() -> Result<()> {
     let start = std::time::Instant::now();
     let dtype = if args.use_f32 {
         DType::F32
-    } else {
+    } else if device.support_native_bf16()? {
```

Comment:
I think we should just switch that `args` into an actual enum `--dtype float16`.